integrations/opennms-snmp-hardware-inventory-provisioning-adapter/src/main/java/org/opennms/netmgt/provision/SnmpHardwareInventoryProvisioningAdapter.java

```diff
@@ -217,19 +218,13 @@ private void synchronizeInventory(int nodeId) {
      */
     private void initializeVendorAttributes() {
         m_vendorAttributes.clear();
-        for (HwEntityAttributeType type : m_hwEntityAttributeTypeDao.findAll()) {
-            LOG.debug("Loading attribute type {}", type);
-            m_vendorAttributes.put(type.getSnmpObjId(), type);
-        }
+        m_hwEntityAttributeTypeDao.deleteAllEntities((Collection<HwEntityAttributeType>) m_hwEntityAttributeTypeDao.findAll());
```

Comment:
Why are we deleting the entities here ?

Reply:
Deleting logic removed. New implementation compares  existing values of hwentityattributetype and values coming from snmp-hardware-inventory-adapter-configuration , if value is not in table and new entry in xml then insert record in table.
If value is already in table then update it with current information (oid, alias, class) coming from xml file.